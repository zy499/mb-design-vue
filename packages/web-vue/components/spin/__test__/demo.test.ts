import demoTest from '../../../scripts/demo-test';

demoTest('spin');
