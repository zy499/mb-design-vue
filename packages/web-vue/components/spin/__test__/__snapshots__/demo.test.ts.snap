// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<spin> demo: render [basic] correctly 1`] = `
"<div class=\\"mb-spin\\">
  <div class=\\"mb-spin-icon\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"mb-icon mb-icon-loading mb-icon-spin\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\">
      <path d=\\"M42 24c0 9.941-8.059 18-18 18S6 33.941 6 24 14.059 6 24 6\\"></path>
    </svg></div>
  <!---->
</div>"
`;

exports[`<spin> demo: render [container] correctly 1`] = `
"<div class=\\"mb-spin mb-spin-loading\\">
  <div class=\\"mb-card mb-card-size-medium mb-card-bordered\\">
    <div class=\\"mb-card-header\\">
      <div class=\\"mb-card-header-title\\">Mb Card</div>
      <!---->
    </div>
    <!---->
    <div class=\\"mb-card-body\\"> ByteDance's core product, Toutiao ('Headlines'), is a content platform in China and around the world. Toutiao started out as a news recommendation engine and gradually evolved into a platform delivering content in various formats.
      <!---->
    </div>
  </div>
  <div class=\\"mb-spin-mask\\">
    <div class=\\"mb-spin-mask-icon\\">
      <div class=\\"mb-spin-icon\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"mb-icon mb-icon-loading mb-icon-spin\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\">
          <path d=\\"M42 24c0 9.941-8.059 18-18 18S6 33.941 6 24 14.059 6 24 6\\"></path>
        </svg></div>
      <div class=\\"mb-spin-tip\\">This may take a while...</div>
    </div>
  </div>
</div>"
`;

exports[`<spin> demo: render [dot] correctly 1`] = `
"<div class=\\"mb-spin\\">
  <div class=\\"mb-spin-icon\\">
    <div class=\\"mb-dot-loading\\">
      <div class=\\"mb-dot-loading-item\\"></div>
      <div class=\\"mb-dot-loading-item\\"></div>
      <div class=\\"mb-dot-loading-item\\"></div>
      <div class=\\"mb-dot-loading-item\\"></div>
      <div class=\\"mb-dot-loading-item\\"></div>
    </div>
  </div>
  <!---->
</div>"
`;

exports[`<spin> demo: render [icon] correctly 1`] = `
"<div class=\\"mb-spin\\">
  <div class=\\"mb-spin-icon\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"mb-icon mb-icon-sync mb-icon-spin\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\">
      <path d=\\"M11.98 11.703c-6.64 6.64-6.64 17.403 0 24.042a16.922 16.922 0 0 0 8.942 4.7M34.603 37.156l1.414-1.415c6.64-6.639 6.64-17.402 0-24.041A16.922 16.922 0 0 0 27.075 7M14.81 11.982l-1.414-1.414-1.414-1.414h2.829v2.828ZM33.192 36.02l1.414 1.414 1.414 1.415h-2.828V36.02Z\\"></path>
    </svg></div>
  <!---->
</div>"
`;

exports[`<spin> demo: render [size] correctly 1`] = `
"<div class=\\"mb-space mb-space-horizontal mb-space-align-center\\">
  <div class=\\"mb-space-item\\" style=\\"margin-right: 24px;\\">
    <div class=\\"mb-spin\\">
      <div class=\\"mb-spin-icon\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"mb-icon mb-icon-loading mb-icon-spin\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\">
          <path d=\\"M42 24c0 9.941-8.059 18-18 18S6 33.941 6 24 14.059 6 24 6\\"></path>
        </svg></div>
      <!---->
    </div>
  </div>
  <div class=\\"mb-space-item\\" style=\\"margin-right: 24px;\\">
    <div class=\\"mb-spin\\">
      <div class=\\"mb-spin-icon\\" style=\\"font-size: 28px;\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"mb-icon mb-icon-loading mb-icon-spin\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\">
          <path d=\\"M42 24c0 9.941-8.059 18-18 18S6 33.941 6 24 14.059 6 24 6\\"></path>
        </svg></div>
      <!---->
    </div>
  </div>
  <div class=\\"mb-space-item\\">
    <div class=\\"mb-spin\\">
      <div class=\\"mb-spin-icon\\" style=\\"font-size: 32px;\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"mb-icon mb-icon-loading mb-icon-spin\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\">
          <path d=\\"M42 24c0 9.941-8.059 18-18 18S6 33.941 6 24 14.059 6 24 6\\"></path>
        </svg></div>
      <!---->
    </div>
  </div>
</div>"
`;

exports[`<spin> demo: render [tip] correctly 1`] = `
"<div class=\\"mb-spin mb-spin-with-tip\\">
  <div class=\\"mb-spin-icon\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"mb-icon mb-icon-loading mb-icon-spin\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\">
      <path d=\\"M42 24c0 9.941-8.059 18-18 18S6 33.941 6 24 14.059 6 24 6\\"></path>
    </svg></div>
  <div class=\\"mb-spin-tip\\">This may take a while...</div>
</div>"
`;
